coral-spark/src/main/java/com/linkedin/coral/spark/CoralSpark.java

```diff
@@ -115,6 +118,43 @@ private static String constructSparkSQL(RelNode sparkRelNode) {
     return rewrittenSparkSqlNode.toSqlString(SparkSqlDialect.INSTANCE).getSql();
   }
 
+  /**
+   * Users use this function as the main API for getting CoralSpark instance.
+   *
+   * Internally Appropriate parts of Sql RelNode is converted to Spark RelNode, Spark RelNode is converted back
+   * to SqlNode and SqlNode to SparkSQL.
+   *
+   * It returns an instance of CoralSpark which contains
+   *  1) Spark SQL
+   *  2) Base tables
+   *  3) Spark UDF information objects, ie. List of {@link SparkUDFInfo}
+   *
+   * @param sqlNode CoralNode which will be translated to SparkSQL.
+   * @param convertor Functional Interface to convert SqlNode to appropriate RelNode
+   *
+   *
+   * @return [[CoralSparkInfo]]
+   */
+  public static CoralSpark create(SqlNode sqlNode, Function<SqlNode, RelNode> convertor){
```

Comment:
If the argument `sqlNode` is of type `SqlCreate`, the query's semantics are not getting validated. For example: 
for a table `foo` with multiple columns, if there are input queries like `create table t1 as select * as col0 from foo` or `create table t1 (a int) as select * from foo`, I'm not sure if Coral will throw an exception right now. 

The validation work can be incremental, but maybe we can setup a class like `ddlSqlValidator` in coral-common / coral-hive modules which can validate the different types of ddl statements and invoke it here. 

We should validate that the rowtype is compatible for `SqlCreateTable`.

Reply:
Queries like `create table t1 as select * as col0 from foo` will fail query parsing, but yes, we would need validation for cases like `create table t1 (a int) as select * from foo`. Do we have anything similar for select queries?

Reply:
for select queries, we use the [hiveSqlValidator](https://github.com/linkedin/coral/blob/master/coral-hive/src/main/java/com/linkedin/coral/hive/hive2rel/HiveSqlValidator.java#L25) to validate the CoralSqlNode and then generate the relNode representation. This is done [here](https://github.com/linkedin/coral/blob/master/coral-hive/src/main/java/com/linkedin/coral/hive/hive2rel/HiveSqlToRelConverter.java#L59).

if create DDL (and other DDL) are not flowing through this translation path, we should validate them separately.

Reply:
@aastha25 I do not think we should create two parallel paths. Can we unify them using common APIs?